lib/jxl/encode.cc

```diff
@@ -157,9 +155,10 @@ JxlEncoderStatus JxlEncoderSetBasicInfo(JxlEncoder* enc,
     return JXL_ENC_ERROR;
   }
   if (info->exponent_bits_per_sample) {
-    if (info->exponent_bits_per_sample != 8) return JXL_ENC_NOT_SUPPORTED;
```

Comment:
we should keep this check, this line would move inside the bits_per_sample==32 case; and for the 16 case we should have this but checking == 5. Those are the only float types supported by this implementation.

Reply:
Ah... yes, moving it would make sense.  I'll make the change.

Reply:
I'm updated the code to make sure 8 is paired with 32 and 5 is paired with 16.